Kudu.Services/Deployment/PushDeploymentController.cs

```diff
@@ -171,33 +335,42 @@ private string GetZipURLFromJSON(JObject requestObject)
                     {
                         throw new ArgumentException("Invalid Url in the JSON request");
                     }
+
                     return packageUri;
                 }
                 catch (Exception ex)
                 {
                     _tracer.TraceError(ex, "Error reading the URL from the JSON {0}", requestObject.ToString());
+
                     throw;
                 }
             }
         }
 
-        private async Task<HttpResponseMessage> PushDeployAsync(ZipDeploymentInfo deploymentInfo, bool isAsync)
+        private async Task<HttpResponseMessage> PushDeployAsync(ArtifactDeploymentInfo deploymentInfo, bool isAsync, JObject requestObject = null, string artifactType = Constants.Zip)
         {
             var content = Request.Content;
             var isRequestJSON = content.Headers?.ContentType?.MediaType?.Equals("application/json", StringComparison.OrdinalIgnoreCase);
             if (isRequestJSON == true)
             {
                 try
                 {
-                    var requestObject = await Request.Content.ReadAsAsync<JObject>();
-                    deploymentInfo.ZipURL = ArmUtils.IsArmRequest(Request) ? GetZipURLFromARMJSON(requestObject) : GetZipURLFromJSON(requestObject);
+                    //Read Httpcontent into a JObject if a JObject does not already exist
+                    if (requestObject == null)
+                    {
+                        requestObject = await Request.Content.ReadAsAsync<JObject>();
+                    }
+                    deploymentInfo.RemoteURL = ArmUtils.IsArmRequest(Request) ? GetArticfactURLFromARMJSON(requestObject) : GetArtifactURLFromJSON(requestObject);
                 }
                 catch (Exception ex)
                 {
                     return ArmUtils.CreateErrorResponse(Request, HttpStatusCode.BadRequest, ex);
                 }
             }
-            else
+            //This is only required for Zip artifacts. It copies the raw ZIP into zipFilePath from where 
```

Comment:
// For zip artifacts (zipdeploy, wardeploy, onedeploy with type=zip), copy the request body in a temp zip file. 
// It will be extracted later as part of the next steps